best-practice-haproxy.md

```diff
@@ -0,0 +1,208 @@
+---
+title: TiDB 最佳实践系列（六）HAProxy 的使用
+author: ['李仲舒']
+date: 2019-11-19
+summary: 本文将从 Java 数据库交互组件开发的角度出发，介绍各组件的推荐配置和推荐使用方式，希望能帮助 Java 开发者在使用 TiDB 时能更好的发挥数据库性能。
+tags: ['HAProxy','最佳实践','TiDB']
+---
+
+HAProxy 是一个使用 C 语言编写的自由及开放源代码软件，其提供高可用性、负载均衡，以及基于 TCP 和 HTTP 的应用程序代理。GitHub、Bitbucket、Stack Overflow、Reddit、Tumblr、Twitter 和 Tuenti 在内的知名网站，及亚马逊网络服务系统都在使用 HAProxy。
+
+TiDB Server 作为无限水平扩展的无状态计算节点，需要像 HAProxy 能提供稳定且高性能的负载均衡组件用对外统一的接口地址来提供服务。HAProxy 在负载均衡的生态中占有很大的市场，TiDB 的用户也可以将这一成熟稳定的开源工具应用的自己的线上业务中承担负载均衡、高可用的功能。
+
+![图 1 部署架构](media/best-practice-haproxy/部署架构.png)
+
+## HAProxy 简介
+
+HAProxy 由 Linux 内核的核心贡献者 Willy Tarreau 于 2000 年编写，他现在仍然负责该项目的维护，并在开源社区免费提供版本迭代。最新的稳定版本 2.0.0 于 2019 年 8 月 16 日发布，带来更多[优秀的特性](https://www.haproxy.com/blog/haproxy-2-0-and-beyond/)。
+
+## HAProxy 部分核心功能
+
+*   [高可用性](http://cbonte.github.io/haproxy-dconv/1.9/intro.html#3.3.4)：HAProxy 提供优雅关闭服务和无缝切换的高可用功能；
+
+*   [负载均衡](http://cbonte.github.io/haproxy-dconv/1.9/configuration.html#4.2-balance)：L4（TCP）和 L7（HTTP）负载均衡模式，至少 9 类均衡算法，比如 roundrobin，leastconn，random 等；
+
+*   [健康检查](http://cbonte.github.io/haproxy-dconv/1.9/configuration.html#5.2-check)：对 HAProxy 配置的 HTTP 或者 TCP 模式状态进行检查；
+
+*   [会话保持](http://cbonte.github.io/haproxy-dconv/1.9/intro.html#3.3.6)：在应用程序没有提供会话保持功能的情况下，HAProxy 可以提供该项功能；
+
+*   [SSL](http://cbonte.github.io/haproxy-dconv/1.9/intro.html#3.3.2)：支持 HTTPS 通信和解析；
+
+*   [监控与统计](http://cbonte.github.io/haproxy-dconv/1.9/intro.html#3.3.3)：通过 web 页面可以实时监控服务状态以及具体的流量信息。
+
+
+## HAProxy 部署操作
+
+### 1. 硬件要求 
+
+根据[HAProxy 官方文档]([http://cbonte.github.io/haproxy-dconv/2.0/management.html#1](http://cbonte.github.io/haproxy-dconv/2.0/management.html#1))对 HAProxy 的服务器硬件配置有以下建议（也可以根据负载均衡环境进行实际推算，在此基础上提高服务器配置）：
+
+|硬件资源|最低配置|
+|:---|:---|
+|CPU|2 核，3.5 GHz|
+|内存|16 GB|
+|存储容量|50 GB（SATA 盘）|
+|网卡|万兆网卡|
+
+#### 2. 软件要求
```

Comment:
@husiyu Could  you please change this to h3? Thank you!

Reply:
done